### PR TITLE
#164798201 Email verification after sign up

### PR DESCRIPTION
**Pull request by dorothyas**
**What does this PR do?**

- Enables a user, on signing up, to receive an email with a verification link

**Description of Task to be completed?**

Users should receive an email containing an activation link when they successfully signup.
Users' accounts should be activated when they click on the activation link they received in their email

**How should this be manually tested?**

- Fetch this branch `git fetch origin ft-signup-email-verification-164798204`
- Checkout to the branch `git checkout ft-signup-email-verification-16479820`
- add the necessary environment variables to your .env file
- Run `python manage.py migrate`
- Run `python manage.py runserver`
- Sign up at POST /api/users/register/
- Try signing in using POST /api/users/login/
- If it fails, go to your email and click on the link to verify
- Try signing in again

What are the relevant pivotal tracker stories?

[#164798201](https://www.pivotaltracker.com/n/projects/2320796/stories/164798201)

[Run in Postman](https://app.getpostman.com/run-collection/456b792a1e3e866a0baf)

### CODE REVIEW

File: authors/apps/authentication/tests/test_data.py [hunk 1]
```diff
@@ -81,3 +81,6 @@
         }
       
 empty_login_data_object = {}
+
+invalid_token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6MTIsImV4cCI6MTU1NDcxMTQ5Nn0.SMPyyqsgiuYxRjJkWd_NsAlhc5bWusLrElvhWERr"
```
Comment (ja-odur) [inline authors/apps/authentication/tests/test_data.py]: @dorothyas This can be achieved through mocking.

File: authors/settings.py [hunk 2]
```diff
@@ -154,4 +154,11 @@
         }
     },
 }
+EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_USE_TLS = True
+EMAIL_PORT = 587
+EMAIL_HOST_USER = os.environ.get('EMAIL_HOST_USER')
```
Comment (ja-odur) [inline authors/settings.py]: @dorothyas please use `python-decouple` to get environment variables.

**Review by ja-odur — DISMISSED** [review]
LGTM

File: authors/apps/authentication/migrations/0002_user_is_verified.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.2 on 2019-04-05 19:58
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
```
Comment (dannylwe) [inline authors/apps/authentication/migrations/0002_user_is_verified.py]: @dorothyas kindly make your migrations into one file

**Review by felixkiryowa — COMMENTED** [review]
Dorothy try raising your coverage it has dropped, however its nice work

**Review by felixkiryowa — APPROVED** [review]

**Review by ezrogha — APPROVED** [review]
LGTM

**Review by ja-odur — APPROVED** [review]
LGTM

**Pull request merged**